PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:3 + add:0, change matches:
-    _inherit = ["stock.picking.type"]
-    _inherit = ["stock.picking"]
-    _inherit = ["stock.picking.batch"]

Total Changes: 6

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/delivery_stock_picking_batch/models/stock_picking.py
+++ b/addons/delivery_stock_picking_batch/models/stock_picking.py
@@ -6,7 +6,7 @@ from odoo.osv import expression
 
 
 class StockPickingType(models.Model):
-    _inherit = ["stock.picking.type"]
+    _inherit = "stock.picking.type"
 
     def _get_default_weight_uom(self):
         return self.env['product.template']._get_weight_uom_name_from_ir_config_parameter()
@@ -27,7 +27,7 @@ class StockPickingType(models.Model):
 
 
 class StockPicking(models.Model):
-    _inherit = ["stock.picking"]
+    _inherit = "stock.picking"
 
     def _get_possible_pickings_domain(self):
         domain = super()._get_possible_pickings_domain()

--- a/addons/delivery_stock_picking_batch/models/stock_picking_batch.py
+++ b/addons/delivery_stock_picking_batch/models/stock_picking_batch.py
@@ -5,7 +5,7 @@ from odoo import models
 
 
 class StockPickingBatch(models.Model):
-    _inherit = ["stock.picking.batch"]
+    _inherit = "stock.picking.batch"
 
     def _is_picking_auto_mergeable(self, picking):
         """ Verifies if a picking can be safely inserted into the batch without violating auto_batch_constrains.
